File: solutions/Question 283/Solution2.cpp
/*
 * Problem: 283. Move Zeroes
 * Difficulty: Easy
 * Topics: Array, Two Pointers
 * URL: https://leetcode.com/problems/move-zeroes/
 * Runtime: 0 ms
 * Memory: 24.9 MB
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int count = 0;
        vector<int> ans;
        for(int x:nums){
            if (x == 0) count++;
            else ans.push_back(x);
        }
        for(int i = 0;i<count;i++){
            ans.push_back(0);
        }
        nums = ans;;
        
    }
};

File: solutions/Question 88/Solution2.py
# Problem: 88. Merge Sorted Array
# Difficulty: Easy
# Topics: Array, Two Pointers, Sorting
# URL: https://leetcode.com/problems/merge-sorted-array/
# Runtime: 1 ms
# Memory: 18 MB
# Solved via CodePath Auto-Committer

class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        nums1[:m] = nums1[:m]  # optional, just for clarity
        # Merge nums1[:m] and nums2, and sort
        nums1[:m+n] = sorted(nums1[:m] + nums2)